freeze_room/__init__.py

```diff
@@ -73,11 +87,11 @@ async def check_event_allowed(
             and event.membership == Membership.LEAVE
             and event.is_state()
         ):
-            has_frozen_room = await self._freeze_room_if_last_admin_is_leaving(
+            need_replace = await self._on_room_leave(
                 event, state_events,
             )
             replacement = None
-            if has_frozen_room:
+            if need_replace:
                 replacement = event.get_dict()
```

Comment:
Yes. There's a bug I need to look into (I haven't made an issue yet because I don't have enough info about it) that causes some state events sent by a module to sometimes be removed from Synapse's db unless we do this to trick Synapse into recalculating the context for the event, which happens if we return an event dict (modified or not).

Reply:
This is https://github.com/matrix-org/synapse/issues/10830. As the issue states I can't reproduce it anymore, so I'm just going to not make it return its own dict.